tests/php/Service/RegisteredMethodManagerTest.php

```diff
@@ -74,4 +76,34 @@ public function testRegisterForMemberDoesNothingWithNoData()
         RegisteredMethodManager::singleton()->registerForMember($member, $method, []);
         $this->assertCount(0, $member->RegisteredMFAMethods());
     }
+
+    public function testDeleteFromMember()
+    {
+        /** @var Member&MemberExtension $member */
+        $member = $this->objFromFixture(Member::class, 'bob_jones');
+
+        // Bob has 3 methods
+        $this->assertCount(3, $member->RegisteredMFAMethods());
+
+        $manager = RegisteredMethodManager::singleton();
+        $manager->deleteFromMember($member, new BasicMathMethod());
+
+        $this->assertCount(2, $member->RegisteredMFAMethods());
+        $this->assertNull($manager->getFromMember($member, new BasicMathMethod()));
+    }
+
+    public function testDeletingLastMethodRemovesBackupCodes()
+    {
+        // Assert the config is set for backup codes
+        Config::modify()->set(MethodRegistry::class, 'default_backup_method', BackupCodeMethod::class);
+
+        /** @var Member&MemberExtension $member */
+        $member = $this->objFromFixture(Member::class, 'jane_doe');
+
+        $this->assertCount(2, $member->RegisteredMFAMethods());
+
+        RegisteredMethodManager::singleton()->deleteFromMember($member, new BasicMathMethod());
+
+        $this->assertCount(0, $member->RegisteredMFAMethods());
```

Comment:
Might want to assert here that the RegisteredMFAMethod is actually being destroyed too.

Reply:
How do you mean? I'm not fetching a method before this.

Reply:
This was intended to compliment the other comment I posted - if this test was verifying that the methods that have been 'deleted' here no longer existed in the DB, it would have flagged the issue during development. It might be a little over the top though? I'll leave the final call up to you 👍

Reply:
I see... Yeah it's weird having the orphan records around - they're essentially inert in the DB... I'll update the test.